datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -295,6 +296,8 @@ fn adjust_input_keys_ordering(
                         new_conditions.0,
                         filter.clone(),
                         join_type,
+                        // TODO: although projection is not used in the join here, because project pushdow is the last rule. Maybe we need to handle it later. Same as filter.
```

Comment:
```suggestion
                        // TODO: although projection is not used in the join here, because projection pushdown is the last rule. Maybe we need to handle it later. Same as filter.
```